examples/ir_benchmarking/colbert/data/collection.py

```diff
@@ -0,0 +1,100 @@
+
+# Could be .tsv or .json. The latter always allows more customization via optional parameters.
+# I think it could be worth doing some kind of parallel reads too, if the file exceeds 1 GiBs.
+# Just need to use a datastructure that shares things across processes without too much pickling.
+# I think multiprocessing.Manager can do that!
+
+import os
+import itertools
+
+from colbert.evaluation.loaders import load_collection
+from colbert.infra.run import Run
+
+
+class Collection:
+    def __init__(self, path=None, data=None):
+        self.path = path
+        self.data = data or self._load_file(path)
+
+    def __iter__(self):
+        # TODO: If __data isn't there, stream from disk!
```

Comment:
Are these really TODOs or they're done by now? Otherwise they sound like tech debt to me.